[2.5.0] Center of charge test failures on macOS runners

## Expected behavior ##

All tests on our latest production release should pass across expected machine types (especially ubuntu, and macos).

## Actual behavior ##

As reported by @orbeckst [here](https://groups.google.com/g/mdnalysis-devel/c/jTgO2eXmMKg/m/NifnvlsxBwAJ?utm_medium=email&utm_source=footer) (and reproduced in #4210), tests for CoC unwrapping are failing.

This isn't seen on develop.

## Code to reproduce the behavior ##

Run tests on the 2.5.0 on an intel macOS machine

On x86_64 intel macOS 13.5 with Python 3.10 and the `-msse3` flag I still get the same failures.

```diff
diff --git a/package/setup.cfg b/package/setup.cfg
index b4151d55f..18ca2f399 100644
--- a/package/setup.cfg
+++ b/package/setup.cfg
@@ -9,7 +9,7 @@
 ## Uncomment the below line to generate the annotated cython HTML output
 # annotate_cython=True
 ## any additional compiler flags for core C routines, excluding ENCORE
-#extra_cflags =
+extra_cflags = -msse3

 [build_sphinx]
 all-files = 1
```
```
$ clang --version
Apple clang version 14.0.3 (clang-1403.0.22.14.1)
Target: x86_64-apple-darwin22.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

## set up and testing

```bash
mamba create -c conda-forge -n mda310y python==3.10 pytest pytest-xdist
mamba activate mda310y
pip install -v -e package/
pip install -v -e testsuite/
```
Inspecting the output shows that
- Cython 3.0 is used (and complains about deprecation of DEF)
- `-msse3` shows up in `clang` lines

run tests
```
pytest --disable-pytest-warnings --pyargs MDAnalysisTests -k Testcenter_of_charge
```

## results

```
(mda310y) deathstar:mdanalysis oliver$ pytest --disable-pytest-warnings --pyargs MDAnalysisTests -k Testcenter_of_charge
======================================================= test session starts ========================================================
platform darwin -- Python 3.10.0, pytest-7.4.0, pluggy-1.2.0
rootdir: ~/MDAnalysis/mdanalysis
plugins: hypothesis-6.82.0, xdist-3.3.1
collected 19037 items / 19022 deselected / 7 skipped / 15 selected

testsuite/MDAnalysisTests/core/test_topologyattrs.py ..........FFFFF                                                         [100%]

============================================================= FAILURES =============================================================
___________________________________________ Testcenter_of_charge.test_coc_unwrap[group] ____________________________________________

self = <MDAnalysisTests.core.test_topologyattrs.Testcenter_of_charge object at 0x130740430>, u = <Universe with 4 atoms>
compound = 'group'

    @pytest.mark.parametrize('compound', compounds)
    def test_coc_unwrap(self, u, compound):
        u.atoms.wrap
        coc = u.atoms[:2].center_of_charge(compound=compound, unwrap=True)
>       assert_equal(coc.flatten(), [0, -0.5, 0])

~/MDAnalysis/mdanalysis/testsuite/MDAnalysisTests/core/test_topologyattrs.py:693:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = (<built-in function eq>, array([0.        , 0.49999997, 0.        ]), [0, -0.5, 0])
kwds = {'err_msg': '', 'header': 'Arrays are not equal', 'strict': False, 'verbose': True}

    @wraps(func)
    def inner(*args, **kwds):
        with self._recreate_cm():
>           return func(*args, **kwds)
E           AssertionError:
E           Arrays are not equal
E
E           Mismatched elements: 1 / 3 (33.3%)
E           Max absolute difference: 0.99999997
E           Max relative difference: 1.99999994
E            x: array([0. , 0.5, 0. ])
E            y: array([ 0. , -0.5,  0. ])

~/tmp/micromamba-root/envs/mda310y/lib/python3.10/contextlib.py:79: AssertionError
__________________________________________ Testcenter_of_charge.test_coc_unwrap[segments] __________________________________________

self = <MDAnalysisTests.core.test_topologyattrs.Testcenter_of_charge object at 0x1307405b0>, u = <Universe with 4 atoms>
compound = 'segments'

    @pytest.mark.parametrize('compound', compounds)
    def test_coc_unwrap(self, u, compound):
        u.atoms.wrap
        coc = u.atoms[:2].center_of_charge(compound=compound, unwrap=True)
>       assert_equal(coc.flatten(), [0, -0.5, 0])

~/MDAnalysis/mdanalysis/testsuite/MDAnalysisTests/core/test_topologyattrs.py:693:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = (<built-in function eq>, array([0.        , 0.49999997, 0.        ]), [0, -0.5, 0])
kwds = {'err_msg': '', 'header': 'Arrays are not equal', 'strict': False, 'verbose': True}

    @wraps(func)
    def inner(*args, **kwds):
        with self._recreate_cm():
>           return func(*args, **kwds)
E           AssertionError:
E           Arrays are not equal
E
E           Mismatched elements: 1 / 3 (33.3%)
E           Max absolute difference: 0.99999997
E           Max relative difference: 1.99999994
E            x: array([0. , 0.5, 0. ])
E            y: array([ 0. , -0.5,  0. ])

~/tmp/micromamba-root/envs/mda310y/lib/python3.10/contextlib.py:79: AssertionError
__________________________________________ Testcenter_of_charge.test_coc_unwrap[residues] __________________________________________

self = <MDAnalysisTests.core.test_topologyattrs.Testcenter_of_charge object at 0x130740640>, u = <Universe with 4 atoms>
compound = 'residues'

    @pytest.mark.parametrize('compound', compounds)
    def test_coc_unwrap(self, u, compound):
        u.atoms.wrap
        coc = u.atoms[:2].center_of_charge(compound=compound, unwrap=True)
>       assert_equal(coc.flatten(), [0, -0.5, 0])

~/MDAnalysis/mdanalysis/testsuite/MDAnalysisTests/core/test_topologyattrs.py:693:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = (<built-in function eq>, array([0.        , 0.49999997, 0.        ]), [0, -0.5, 0])
kwds = {'err_msg': '', 'header': 'Arrays are not equal', 'strict': False, 'verbose': True}

    @wraps(func)
    def inner(*args, **kwds):
        with self._recreate_cm():
>           return func(*args, **kwds)
E           AssertionError:
E           Arrays are not equal
E
E           Mismatched elements: 1 / 3 (33.3%)
E           Max absolute difference: 0.99999997
E           Max relative difference: 1.99999994
E            x: array([0. , 0.5, 0. ])
E            y: array([ 0. , -0.5,  0. ])

~/tmp/micromamba-root/envs/mda310y/lib/python3.10/contextlib.py:79: AssertionError
_________________________________________ Testcenter_of_charge.test_coc_unwrap[molecules] __________________________________________

self = <MDAnalysisTests.core.test_topologyattrs.Testcenter_of_charge object at 0x1307406d0>, u = <Universe with 4 atoms>
compound = 'molecules'

    @pytest.mark.parametrize('compound', compounds)
    def test_coc_unwrap(self, u, compound):
        u.atoms.wrap
        coc = u.atoms[:2].center_of_charge(compound=compound, unwrap=True)
>       assert_equal(coc.flatten(), [0, -0.5, 0])

~/MDAnalysis/mdanalysis/testsuite/MDAnalysisTests/core/test_topologyattrs.py:693:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = (<built-in function eq>, array([0.        , 0.49999997, 0.        ]), [0, -0.5, 0])
kwds = {'err_msg': '', 'header': 'Arrays are not equal', 'strict': False, 'verbose': True}

    @wraps(func)
    def inner(*args, **kwds):
        with self._recreate_cm():
>           return func(*args, **kwds)
E           AssertionError:
E           Arrays are not equal
E
E           Mismatched elements: 1 / 3 (33.3%)
E           Max absolute difference: 0.99999997
E           Max relative difference: 1.99999994
E            x: array([0. , 0.5, 0. ])
E            y: array([ 0. , -0.5,  0. ])

~/tmp/micromamba-root/envs/mda310y/lib/python3.10/contextlib.py:79: AssertionError
_________________________________________ Testcenter_of_charge.test_coc_unwrap[fragments] __________________________________________

self = <MDAnalysisTests.core.test_topologyattrs.Testcenter_of_charge object at 0x130740760>, u = <Universe with 4 atoms>
compound = 'fragments'

    @pytest.mark.parametrize('compound', compounds)
    def test_coc_unwrap(self, u, compound):
        u.atoms.wrap
        coc = u.atoms[:2].center_of_charge(compound=compound, unwrap=True)
>       assert_equal(coc.flatten(), [0, -0.5, 0])

~/MDAnalysis/mdanalysis/testsuite/MDAnalysisTests/core/test_topologyattrs.py:693:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = (<built-in function eq>, array([0.        , 0.49999997, 0.        ]), [0, -0.5, 0])
kwds = {'err_msg': '', 'header': 'Arrays are not equal', 'strict': False, 'verbose': True}

    @wraps(func)
    def inner(*args, **kwds):
        with self._recreate_cm():
>           return func(*args, **kwds)
E           AssertionError:
E           Arrays are not equal
E
E           Mismatched elements: 1 / 3 (33.3%)
E           Max absolute difference: 0.99999997
E           Max relative difference: 1.99999994
E            x: array([0. , 0.5, 0. ])
E            y: array([ 0. , -0.5,  0. ])

~/tmp/micromamba-root/envs/mda310y/lib/python3.10/contextlib.py:79: AssertionError
===================================================== short test summary info ======================================================
FAILED testsuite/MDAnalysisTests/core/test_topologyattrs.py::Testcenter_of_charge::test_coc_unwrap[group] - AssertionError:
FAILED testsuite/MDAnalysisTests/core/test_topologyattrs.py::Testcenter_of_charge::test_coc_unwrap[segments] - AssertionError:
FAILED testsuite/MDAnalysisTests/core/test_topologyattrs.py::Testcenter_of_charge::test_coc_unwrap[residues] - AssertionError:
FAILED testsuite/MDAnalysisTests/core/test_topologyattrs.py::Testcenter_of_charge::test_coc_unwrap[molecules] - AssertionError:
FAILED testsuite/MDAnalysisTests/core/test_topologyattrs.py::Testcenter_of_charge::test_coc_unwrap[fragments] - AssertionError:
============================== 5 failed, 10 passed, 7 skipped, 19022 deselected, 6 warnings in 27.97s ==============================
```

--- a/package/CHANGELOG
+++ b/package/CHANGELOG
@@ -21,9 +21,9 @@ The rules for this file:
  * 2.6.0
 
 Fixes
-  * Fix AtomGroup.center_of_charge(..., unwrap=True) giving wrong
-    results on Intel macOS by removing `-ffast-math` compiler flag
-    (Issues #3821, #4211)    
+  * Fix AtomGroup.center_of_charge(..., unwrap=True) giving
+    inconsistent (but scientifically correct) results on Intel macOS
+    (Issue #4211)
   * Fix Boltzmann typo in `units.py` (PR #4214, Issue #4213)
   * Fix deletion of bond/angle/dihedral types (PR #4003, Issue #4001)
   * CRD, PQR, and PDBQT writers now write BZ2 and GZ files if requested
@@ -35,6 +35,11 @@ Enhancements
     (Issue #3546)
 
 Changes
+  * Removed `-ffast-math` compiler flag to avoid potentially incorrect
+    floating point results in MDAnalysis *and* other codes when
+    MDAnalysis shared libraries are loaded --- see 
+    https://moyix.blogspot.com/2022/09/someones-been-messing-with-my-subnormals.html
+    (Issues #3821, #4211)
   * Atom ID representation order updated to match that of associated
     AtomGroup indices.
     (PR #4191, Issue #4181)

--- a/package/CHANGELOG
+++ b/package/CHANGELOG
@@ -14,11 +14,16 @@ The rules for this file:
 
 ------------------------------------------------------------------------------
 
-??/??/?? IAlibay, jaclark5, MohitKumar020291, orionarcher, xhgchen, ztimol
+??/??/?? IAlibay, jaclark5, MohitKumar020291, orionarcher, xhgchen,
+         ztimol, orbeckst
+
 
  * 2.6.0
 
 Fixes
+  * Fix AtomGroup.center_of_charge(..., unwrap=True) giving wrong
+    results on Intel macOS by removing `-ffast-math` compiler flag
+    (Issues #3821, #4211)    
   * Fix Boltzmann typo in `units.py` (PR #4214, Issue #4213)
   * Fix deletion of bond/angle/dihedral types (PR #4003, Issue #4001)
   * CRD, PQR, and PDBQT writers now write BZ2 and GZ files if requested

--- a/package/setup.py
+++ b/package/setup.py
@@ -268,7 +268,7 @@ def extensions(config):
     use_openmp = config.get('use_openmp', default=True)
     annotate_cython = config.get('annotate_cython', default=False)
 
-    extra_compile_args = ['-std=c99', '-ffast-math', '-O3', '-funroll-loops',
+    extra_compile_args = ['-std=c99', '-O3', '-funroll-loops',
                           '-fsigned-zeros'] # see #2722
     define_macros = []
     if config.get('debug_cflags', default=False):

